polus-training-splinedist-plugin/src/utils.py

```diff
@@ -0,0 +1,520 @@
+    average_jaccard = format(average_jaccard, ".3f")
+    standard_deviation_jaccard = np.std(jaccard_indexes)
+    standard_deviation_jaccard = format(standard_deviation_jaccard, ".3f")
+    logger.info("Average Jaccard Index for Testing Data: {} ± {}".format(average_jaccard, standard_deviation_jaccard))
+
+def train_nn(image_dir_input : str,
+             label_dir_input : str,
+             image_dir_test : str,
+             label_dir_test : str,
+             split_percentile : int,
+             output_directory : str,
+             gpu : bool,
+             imagepattern : str,
+             M : int,
+             epochs : int):
+
+    """ This function either trains or continues to train a neural network 
+    for SplineDist.
+    Either testing directories are specified or the the input directory gets 
+    split into training and testing data. 
+
+    Args:
+        image_dir_input: location for Intensity Based Images
+        label_dir_input: location for Ground Truths of the images
+        image_dir_test: Specifies the location for Intensity Based Images for testing
+        label_dir_test: Specifies the location for Ground truth images for testing
+        split_percentile: Specifies what percentages of the input should be allocated for tested
+        output_directory: Specifies the location for the output generated
+        gpu: Specifies whether or not there is a GPU to use
+        imagepattern: The imagepattern of files to iterate through within a directory
+        M: Specifies the number of control points
+        epochs : Specifies the number of epochs to be run
+
+    Returns:
+        None, a trained neural network whose performance is calculated by the jaccard index
+    
+    Raises:
+        AssertionError: If there is less than one training image
+        AssertionError: If the number of images to do not match the number of ground truths 
+                        available.
+    """
+
+    assert isinstance(M, int), "Neeed to specify the number of control points"
+    assert isinstance(epochs, int), "Need to specify the number of epochs to run"
+
+    # get the inputs
+    input_images = []
+    input_labels = []
+    for files in fp(image_dir_input,imagepattern)():
+        image = files[0]['file']
+        if os.path.exists(image):
+            input_images.append(image)
+    for files in fp(label_dir_input,imagepattern)():
+        label = files[0]['file']
+        if os.path.exists(label):
+            input_labels.append(label)
+
+    input_images = sorted(input_images)
+    input_labels = sorted(input_labels)
+    num_inputs = len(input_images)
+    
+    X_trn = []
+    Y_trn = []
+    X_val = []
+    Y_val = []
+
+    logger.info("\n Getting Data for Training and Testing  ...")
+    split_percentile = int(split_percentile)
+    logger.info("Split Percentile {} ({})".format(split_percentile, type(split_percentile)))
+    if (split_percentile == None) or int(split_percentile) == 0:
```

Comment:
You should do `split_percentile is None`. It shouldn't matter much here, but this checks for whether the variables point to the same object rather than checking for value equality.